### PR TITLE
Migrate Flexport Source Connector to Low Code

### LINKED ISSUE #28995: flexport
Please use the [connector builder](https://docs.airbyte.com/connector-development/config-based/low-code-cdk-overview/?_gl=1*13hxonw*_gcl_au*MTI0NTMzNjM3OS4xNjkwODM4MzE0*_ga*MTg2NTc3OTc0Ni4xNjUxNTk0NTQw*_ga_HDBMVFQGBH*MTY5MTAwNDEzMy4yNzIuMC4xNjkxMDA0MTMzLjAuMC4w&_ga=2.197654767.1978878.1690818922-1865779746.1651594540) for this.

### PR DESCRIPTION
**Pull request by topefolorunso**
## What
Migrating Source Flexport to Low-Code CDK resolves #28995

## How
Created the connector using the connector buider UI, then imported the manifest.yaml to the low code cdk.

## Recommended reading order
1. `manifest.yaml`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-flexport/source_flexport/schemas/TODO.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# TODO: Define your stream schemas
+Your connector must describe the schema of each stream it can output using [JSONSchema](https://json-schema.org). 
+
+You can describe the schema of your streams using one `.json` file per stream.
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-flexport/source_flexport/schemas/TODO.md:4]: Please remove this file.

File: airbyte-integrations/connectors/source-flexport/metadata.yaml [hunk 2]
```diff
@@ -1,23 +1,28 @@
 data:
+  allowedHosts:
+    hosts:
+      - TODO # Please change to the hostname of the source.
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-flexport/metadata.yaml:4]: Please add the allowed hosts here.

File: airbyte-integrations/connectors/source-flexport/metadata.yaml [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
+  registries:
+    oss:
+      enabled: true
+    cloud:
+      enabled: false
   connectorSubtype: api
   connectorType: source
   definitionId: f95337f1-2ad1-4baf-922f-2ca9152de630
-  dockerImageTag: 0.1.0
+  dockerImageTag: 0.4.0
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-flexport/metadata.yaml:13]: Why are you bumping 3 minor versions and not only one?

File: airbyte-integrations/connectors/source-flexport/source_flexport/manifest.yaml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      start_date:
+        type: string
+        title: Start date
+        format: date-time
+        pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$
+        order: 1
+
+    additionalProperties: true
+  documentation_url: "https://docs.airbyte.com/integrations/sources/flexport"
+  type: Spec
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-flexport/source_flexport/manifest.yaml:158]: Can you format this?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments. Can you share the output from integration tests?

**Comment by topefolorunso** [comment]
> Some comments. Can you share the output from integration tests?

@marcosmarxm I couldn't get credentials from Flexport API for testing. Perhaps you can help with that.

File: airbyte-integrations/connectors/source-flexport/source_flexport/manifest.yaml (hunk 4 above)
Comment (topefolorunso) [reply airbyte-integrations/connectors/source-flexport/source_flexport/manifest.yaml:158]: Can you clarify what you mean by format?

**Comment by marcosmarxm** [comment]
> 

@topefolorunso please send a msg to @RealChrisSean in Slack

File: airbyte-integrations/connectors/source-flexport/source_flexport/manifest.yaml (hunk 4 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-flexport/source_flexport/manifest.yaml:158]:
```yaml
spec:
  type: Spec
  documentation_url: "https://docs.airbyte.com/integrations/sources/flexport"
  connection_specification:
    $schema: http://json-schema.org/draft-07/schema#
    type: object
    additionalProperties: true
    required:
      - api_key
      - start_date
    properties:
      api_key:
        type: string
        title: API Key
        airbyte_secret: true
        order: 0
      start_date:
        type: string
        title: Start date
        format: date-time
        pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$
        order: 1

File: airbyte-integrations/connectors/source-flexport/metadata.yaml (hunk 3 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-flexport/metadata.yaml]: https://semver.org/

Comment (topefolorunso) [reply airbyte-integrations/connectors/source-flexport/metadata.yaml]: > [semver.org](https://semver.org/)

Thank you!

**Comment by marcosmarxm** [comment]
Hello @topefolorunso the implementation looks good. Let's wait for the credentials to run integration tests.

**Review by marcosmarxm — APPROVED** [review]
Thanks for the migration @topefolorunso the new connector has the same implementation as the previous version.

**Pull request merged**